node/ChainwebNode.hs

```diff
@@ -452,16 +450,7 @@ mainInfo = programInfoValidate
     (defaultChainwebNodeConfiguration Mainnet01)
     validateChainwebNodeConfiguration
 
--- | KILLSWITCH: The logic surrounding `txSilenceEndDate` here is to be removed in
--- a future version of Chainweb. This prevents the Node from even starting if
--- past a specified date.
---
 main :: IO ()
 main = withWatchdog . runWithPkgInfoConfiguration mainInfo pkgInfo $ \conf -> do
     let v = _configChainwebVersion $ _nodeConfigChainweb conf
-    now <- getCurrentTimeIntegral
-    case txSilenceEndDate v of
-        Just end | now > end -> do
-            putStrLn "Transactions are now possible - please update your Chainweb binary."
-            exitFailure
-        _ -> withNodeLogger (_nodeConfigLog conf) v $ node conf
+    withNodeLogger (_nodeConfigLog conf) v $ node conf
```

Comment:
We need a new global kill switch here, that shuts off the node completely (not only preventing restarts).

I propose that we implement something like `withKillSwitch :: Maybe Date -> IO a -> IO a` and use it at the top level in `main`. For 1.1. it must be set to `2020-01-15T00:00:00Z`, according to the overall timeline.

Should that be done in this PR or in a follow up PR?

Reply:
Follow-up (Part 3).